scripts/_common.sh

```diff
@@ -1,69 +1,67 @@
-### Constants
+#!/bin/bash
 
-supervisor_conf_path="/etc/supervisor/conf.d/ihatemoney.conf"
-gunicorn_conf_path="/etc/ihatemoney/gunicorn.conf.py"
-ihatemoney_conf_path="/etc/ihatemoney/ihatemoney.cfg"
-INSTALL_DIR="/opt/yunohost/ihatemoney"
+#=================================================
+# COMMON VARIABLES
+#=================================================
 
+# dependencies used by the app
+pkg_dependencies=(
+    python3-dev
+    python3-virtualenv
+    libffi-dev
+    libssl-dev
+    virtualenv
+)
 
-### Functions
+pip_dependencies=(
+    'setuptools>=18.5'
+    'gunicorn>=19.3.0'
+    'PyMySQL>=0.9,<0.10'
+    'SQLAlchemy<1.4'
+    'ihatemoney>=4,<5'
+)
 
+### Constants
 
-install_apt_dependencies() {
-    ynh_install_app_dependencies \
-        python3-dev \
-        python3-virtualenv \
-        libffi-dev \
-        libssl-dev \
-        supervisor \
-        virtualenv
-}
+#=================================================
+# PERSONAL HELPERS
+#=================================================
 
-create_unix_user() {
-    mkdir -p /opt/yunohost
-    useradd ihatemoney -d /opt/yunohost/ihatemoney/ --create-home || ynh_die "User creation failed"
-}
+__ynh_python_virtualenv_setup() {
+    local -A args_array=( [d]=venv_dir= [p]=packages= )
+    local venv_dir
+    local packages
+    ynh_handle_getopts_args "$@"
 
-create_system_dirs() {
-    install -o ihatemoney -g ihatemoney -m 755 -d \
-         /var/log/ihatemoney \
-         /etc/ihatemoney
-    mkdir -p /opt/yunohost
-}
+    virtualenv --python=python3 --system-site-packages "$venv_dir"
+
+    if [[ -n "$packages" ]]; then
+        IFS=" " read -r -a pip_packages <<< "$packages"
 
-init_virtualenv () {
-    virtualenv /opt/yunohost/ihatemoney/venv --python /usr/bin/python3
+        (
+            set +o nounset
```

Comment:
I do not realy agree with the use of bin/activate in interactive script, makes the whole thing a big magic and statefull, and here seems to require this extra step with unset/nounset

Reply:
What's the problem with activate ? Here it's encapsulated in a subshell.